src/new/wallet.js

```diff
@@ -184,6 +184,129 @@ class HathorWallet extends EventEmitter {
     return historyTransactions;
   }
 
+  /**
+   * Get utxos of the wallet addresses
+   *
+   * @param {object} options Consolidation options shared with filterUtxos and consolidateUtxos (https://github.com/HathorNetwork/hathor-wallet-headless/issues/44)
+   *
+   * @return {object} { total_amount_available: number, total_utxos_available: number, total_amount_locked: number, total_utxos_locked: number, utxos: Array[] } utxos and meta information about it
+   */
+  getUtxos(options) {
+    storage.setStore(this.store);
+    const historyTransactions = Object.values(this.getTxHistory());
+    const utxoDetails = {
+      total_amount_available: 0,
+      total_utxos_available: 0,
+      total_amount_locked: 0,
+      total_utxos_locked: 0,
+      utxos: [],
+    };
+
+    // Iterate through transactions
+    for (let i = 0; i < historyTransactions.length; i++) {
+      const transaction = historyTransactions[i];
+
+      // Orphan transactions should be ignored
+      if (transaction.is_voided) continue;
+
+      // Iterate through outputs
+      for (let j = 0; j < transaction.outputs.length; j++) {
+        const output = transaction.outputs[j];
+
+        const is_unspent = output.spent_by === null;
+        if (!is_unspent) {
+          // No other checks required
+          continue;
+        }
+
```

Comment:
Shouldn't we check whether it's been used by another transactions that hasn't been received by the websocket yet (see https://github.com/HathorNetwork/hathor-wallet-lib/pull/141)?

Reply:
Hi @msbrogli. Talking with @pedroferreira1 he explained that [`canUseUnspentTx`](https://github.com/HathorNetwork/hathor-wallet-lib/blob/3ab23e174560b6140dedc512995a851739eb2811/src/wallet.js#L997) prevents issues with utxo being used by another transaction. So, as I used `canUseUnspentTx` here, this issue will not happen.

Reply:
Great! I think we should add a note explaining that `canUseUnspentTx()` already handles this case.

Reply:
Done